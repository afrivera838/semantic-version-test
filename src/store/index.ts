import { combineReducers, configureStore } from '@reduxjs/toolkit';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { encryptTransform } from 'redux-persist-transform-encrypt';
import { exampleReducer } from '../modules/example/redux';
import _ from 'lodash';
import storageSession from 'redux-persist/lib/storage/session';

const rootReducer = combineReducers({
  ...exampleReducer,
});

// eslint-disable-next-line no-undef
const secretKey = process.env.REACT_APP_REDUX_KEY || 'test';
const keyName = `${process.env.REACT_APP_NAME}-key`;

const persistConfig = {
  key: keyName,
  storage: storageSession,
  version: 1,
  whitelist: [],
  transforms: [
    encryptTransform({
      secretKey,
      onError: function (error) {
        console.error(error);
      },
    }),
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  // eslint-disable-next-line no-undef
  devTools: process.env.REACT_APP_ENVIRONMENT !== 'prod',
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, PAUSE, REHYDRATE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export const persistor = persistStore(store);
