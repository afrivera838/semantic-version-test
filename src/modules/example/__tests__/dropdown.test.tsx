import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from '../../../store';
import Example from '../Example';

interface Props {
  statusEvents: (a:string) => void;
}

describe('Root component', () => {
  function renderApp(props?: Props) {
    return {
      ...render(
        <BrowserRouter>
          <Example />
        </BrowserRouter>
      ),
    };
  }

  test('should be in the document', () => {
    const { container } = renderApp();
    expect(container).toMatchSnapshot();
    expect(container).toBeInTheDocument();

    /* const { getByText } = render(<Example />);
    expect(getByText(/Hello Boilerplate!/i)).toBeInTheDocument(); */
  });
});
