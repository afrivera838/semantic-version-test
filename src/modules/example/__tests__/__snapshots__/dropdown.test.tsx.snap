// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component should be in the document 1`] = `
<div>
  <div>
    <h1
      class="colorBlue smallText"
    >
      Hello Boilerplate!
    </h1>
  </div>
</div>
`;
